### PR TITLE
Fix a crash when `Sprite3D.extra_cull_margin` is positive and then assign a texture to it

### LINKED ISSUE #80749: Silent crash when Sprite3D.texture is assigned while null and Sprite3D.extra_cull_margin is positive
### Godot version

v4.1.1.stable.official [bd6af8e0e] v4.0.3.stable.official [5222a99f5]

### System information

Godot v4.1.1.stable - Windows 10.0.19044 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1660 SUPER (NVIDIA; 31.0.15.3168) - AMD Ryzen 5 5600G with Radeon Graphics (12 Threads)

### Issue description

Under these circumstances a silent game crash will occur when running a scene while using the editor or in an exported release:
1. `$Sprite3D.extra_cull_margin = 0.25` or any positive number.
2. `$Sprite3D.texture == null`.
3. Crashes upon `$Sprite3D.texture = preload("res://icon.svg")`, or any texture, but only if not in `_ready()`.

Unsure if related, but there is an alternative silent crash if `extra_cull_margin` is set to a large number like 100000000. However, it occurs the second time `$Sprite3D.texture` is assigned to `null` of all things.

When happening in my game, the crash was a bit weirder but I can't recreate it. It was instead randomly crashing over a period of minutes, after when the repro would have crashed. During that time, the Sprite3D would be correctly assigned a texture(checked via remote view and prints), but it would be invisible in game.

Notes:
- I print just before the crashes, and the print is caught by the external console but not the editor console upon crash.
- No crash if assigning the equivalent with `$Sprite3D.custom_aabb`.
- `$Sprite3D.extra_cull_margin = -1.0` will obviously error, so it doesn't crash.
- Crash occurs with all renderers(4.1 and 4.0 tested).
- Reproduced in 4.1 and 4.0, 3.5 acts as expected and is nominal.
- Camera3D is not needed for reproduction, it is only for viewing the Sprite3D.

### Steps to reproduce

Download and run `main.tscn` to see the crash occur. `main.gd` contains commentary and example code.
If you use an external console, you may see one more print compared to the editor's console upon crash.
Uncomment `$Sprite3D.extra_cull_margin = 100000000.0` in `main.gd` to see alternative crash.

### Minimal reproduction project

[crash_reproduction.zip](https://github.com/godotengine/godot/files/12376628/crash_reproduction.zip)

### PR DESCRIPTION
**Pull request by jsjtxietian**
This pr fixes #80749.
It works in my test but I'm no familiar with the code logic so I'm not sure whether this change will break other things. 

The bug's root cause can be found in the issue conversation. Basically what I do is to rebuild the bvh when p_instance's base_type change from `INSTANCE_NONE` to any `INSTANCE_GEOMETRY`. Maybe there is a better way to fix this issue, thank you for your suggestions in advance.

### CODE REVIEW

File: servers/rendering/renderer_scene_cull.cpp [hunk 1]
```diff
@@ -608,8 +608,19 @@ void RendererSceneCull::instance_set_base(RID p_instance, RID p_base) {
 	instance->base = RID();
 
 	if (p_base.is_valid()) {
+		RS::InstanceType old_base = instance->base_type;
 		instance->base_type = RSG::utilities->get_base_type(p_base);
 
+		if (old_base == RS::INSTANCE_NONE && ((1 << instance->base_type) & RS::INSTANCE_GEOMETRY_MASK)
```
Comment (AThousandShips) [inline servers/rendering/renderer_scene_cull.cpp:614]:
```suggestion
		if (old_base == RS::INSTANCE_NONE && ((1 << instance->base_type) & RS::INSTANCE_GEOMETRY_MASK) {
```

File: servers/rendering/renderer_scene_cull.cpp [hunk 2]
```diff
@@ -608,8 +608,19 @@ void RendererSceneCull::instance_set_base(RID p_instance, RID p_base) {
 	instance->base = RID();
 
 	if (p_base.is_valid()) {
+		RS::InstanceType old_base = instance->base_type;
 		instance->base_type = RSG::utilities->get_base_type(p_base);
 
+		if (old_base == RS::INSTANCE_NONE && ((1 << instance->base_type) & RS::INSTANCE_GEOMETRY_MASK) 
+			&& instance->indexer_id.is_valid()) {
```
Comment (AThousandShips) [inline servers/rendering/renderer_scene_cull.cpp:615]:
```suggestion
		if (old_base == RS::INSTANCE_NONE && ((1 << instance->base_type) & RS::INSTANCE_GEOMETRY_MASK) && instance->indexer_id.is_valid()) {
```

Comment (jsjtxietian) [reply servers/rendering/renderer_scene_cull.cpp:615]: Somehow the clang-format in my local machine doesn't always seems to work, sorry. 
I will deal with formatting problems after reviews think this method makes sense.

Comment (jsjtxietian) [reply servers/rendering/renderer_scene_cull.cpp:615]: Thanks！

**Comment by AThousandShips** [comment]
Might solve:
* https://github.com/godotengine/godot/issues/80504

**Comment by jsjtxietian** [comment]
I haven't update this pr for a long time, probably need some re-thinking on the fix now ( back then I had little experience of Godot ).

**Comment by jsjtxietian** [comment]
Superseded by https://github.com/godotengine/godot/pull/95503

**Pull request closed without merging**